Source/CoreDataExtensions.swift

```diff
@@ -199,3 +199,83 @@ public func resetStack(stack: CoreDataStack,
             }
         }
 }
+
+/**
+ Progressively migrates the persistent store of a `CoreDataModel` based on mapping models found in the model's bundle.
+
+ - note: Migration is only supported for on-disk persistent stores.
+  A complete 'path' of mapping models must exist between the peristent store's version and the model's version
+
+ - parameter model:      The `CoreDataModel` to perform migration upon.
+ 
+ - throws: If the `NSManagedObjectModel` corresponding to the existing persistent store is not found in the model bundle then this function throws `.SourceManagedObjectModelNotFound`. If an `NSMappingModel` part of the progressive migration 'path' is not found in the model bundle then a `.MappingModelNotFound` is thrown. The function rethrows exceptions that may result from reading the persistent store metadata, file system operations or migration with `NSMigrationManager`.
+ */
+public func migrate(model: CoreDataModel) throws {
+
+    func findModelsInBundle(bundle: NSBundle) -> [NSManagedObjectModel] {
+        guard let modelDirectoryURLs = bundle.URLsForResourcesWithExtension("momd", subdirectory: nil) else {
+            return []
+        }
+
+        return Array(modelDirectoryURLs.flatMap { $0.lastPathComponent }.flatMap { bundle.URLsForResourcesWithExtension("mom", subdirectory: $0) }.flatten())
+            .flatMap { NSManagedObjectModel(contentsOfURL: $0) }
```

Comment:
let's breakup this return into multiple steps with descriptive variable names so its easier to read :smile: